plugins/citadel-paper/src/main/java/vg/civcraft/mc/citadel/command/Acid.java

```diff
@@ -54,24 +56,33 @@ public void execute(Player p) {
                     "You do not have sufficient permission to use acid blocks on this group.");
                 return;
             }
-            long neededTime = acidMan.getRemainingAcidMaturationTime(reinforcement);
-            if (neededTime > 0) {
-                CitadelUtility.sendAndLog(p, ChatColor.RED, "That acid block will be mature in "
-                    + TextUtil.formatDuration(neededTime, TimeUnit.MILLISECONDS));
-                return;
-            }
 
-            AcidType acidType = acidMan.getAcidTypeFromMaterial(block.getType());
+            Map<BlockFace, Long> remainingTimes = acidMan.getRemainingAcidMaturationTime(reinforcement);
+            Map<BlockFace, Long> filteredEntries = remainingTimes.entrySet().stream()
+                .filter(entry -> entry.getValue() <= 0)
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            for (BlockFace blockFace : remainingTimes.keySet()) {
+                Block relativeBlock = block.getRelative(blockFace);
+                Reinforcement relativeReinforcement = ReinforcementLogic.getReinforcementProtecting(relativeBlock);
+
+                if (filteredEntries.isEmpty())  {
+                    if (relativeReinforcement != null && !acidMan.canAcidBlock(reinforcement.getType(), relativeReinforcement.getType())) {
+                        CitadelUtility.sendAndLog(p, ChatColor.RED, String.format("The acid facing %s will fail!", blockFace));
+                    }
+                    CitadelUtility.sendAndLog(p, ChatColor.RED, String.format("This acid block is not ready yet."));
+                    return;
+                }
+            }
 
-            for (BlockFace blockFace : acidType.blockFaces()) {
+            for (BlockFace blockFace : filteredEntries.keySet()) {
                 Block relativeBlock = block.getRelative(blockFace);
 
                 Reinforcement relativeReinforcement = ReinforcementLogic.getReinforcementProtecting(relativeBlock);
                 if (
                     relativeReinforcement == null
                         || !relativeReinforcement.getType().canBeReinforced(relativeBlock.getType())
                         || !acidMan.canAcidBlock(reinforcement.getType(), relativeReinforcement.getType())
-                        || acidMan.isPossibleAcidBlock(relativeBlock)
+                        || acidMan.isPossibleAcidBlock(relativeBlock) && acidMan.isAcidOnSameGroup(reinforcement, relativeReinforcement)
```

Comment:
misleading formatting, the `&&` is on the same line as the previous check but applies to the whole statement, choose one of these, depending on intent:

`|| (acidMan.isPossibleAcidBlock(relativeBlock) && acidMan.isAcidOnSameGroup(reinforcement, relativeReinforcement))`

`|| acidMan.isPossibleAcidBlock(relativeBlock)
&& acidMan.isAcidOnSameGroup(reinforcement, relativeReinforcement)`

also same nit as before: whole statement can be 1-indented

Reply:
actually, shouldn't it just be
```
|| acidMan.isPossibleAcidBlock(relativeBlock)
|| acidMan.isAcidOnSameGroup(reinforcement, relativeReinforcement)
```
?

was this tested?

Reply:
oh it should be 
```
|| (acidMan.isPossibleAcidBlock(relativeBlock) && acidMan.isAcidOnSameGroup(reinforcement, relativeReinforcement))
```
right?

Reply:
oops yeah that should be what it is